### PR TITLE
Feat/lens

### PR DESCRIPTION
**Pull request by thiagobrez**
### Summary

- Add Lens SDK for web and native. Import conditionally based on Platform-specific files
- After connecting wallet, display screent o sign in with lens, showing available profiles
- Display feed of (simplified) publications on Home screen
- Ability to upvote (like) a publication
- Display basic profile info in Profile screen
- Ability to logout

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,20 +1,27 @@
...
 - A [Wallet Connect Cloud](https://cloud.walletconnect.com/sign-in) project ID
-- An [Alchemy](https://www.alchemy.com/) API key
 - Expo Go app installed in your smartphone
 - One or more web3 wallets installed in your smartphone (e.g. MetaMask, Rainbow Wallet, Trust Wallet, etc)
 
 ## How to run
 
-- Rename `.env.example` to `.env` and fill in your Wallet Connect Cloud project ID, and Alchemy API key
+- Rename `.env.example` to `.env` and fill in your Wallet Connect Cloud project ID
+- `npm install --legacy-peer-deps`
```
Comment (mdjastrzebski) [inline README.md:18]: Q: why use --legacy-per-deps?
Conisder adding short info on why we are using that option to the README

File: src/components/LensButton/index.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import { useLogin } from '@lens-protocol/react-native';
+import { useAccount } from 'wagmi';
+import { ProfileId } from '../../lib/lens-sdk';
+import Button from '../Button';
+
+type LensButtonProps = {
+  profileId: ProfileId;
+};
+
+export function LensButton({ profileId }: LensButtonProps) {
```
Comment (mdjastrzebski) [inline src/components/LensButton/index.tsx:11]: [nit] Add short TSDoc explaining purpose of this button

File: src/features/home/HomeScreen.tsx [hunk 3]
```diff
@@ -1,44 +1,41 @@
...
+  const { data: session } = useSession();
 
-  if (!isConnected) {
-    return (
-      <View style={styles.container}>
-        <Text>Connect wallet to display NFTs</Text>
-      </View>
-    );
-  }
+  const { data, loading } = useFeed({
```
Comment (mdjastrzebski) [inline src/features/home/HomeScreen.tsx:10]: [nit] Consider extracting as `useHomeFeed` or similar

File: src/features/home/PublicationItem.tsx [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+  };
+
+  // For simplicity, filtering out EventMetadataV3 items because they do not have content
+  if (publication.metadata.__typename === 'EventMetadataV3') {
+    return null;
+  }
+
+  return (
+    <View style={styles.container}>
+      <View style={styles.profileContainer}>
```
Comment (mdjastrzebski) [inline src/features/home/PublicationItem.tsx:41]: [nit] Consider breaking it into some smaller meaningful components like Profile, Reactions, etc

File: src/features/sign-in/SignInScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  const { data: profiles = [] } = useProfilesManaged({
+    for: address,
+  });
+
+  // TODO: useLastLoggedInProfile seems to not be updating right after logout
+  const { data: lastLoggedInProfile } = useLastLoggedInProfile({
+    for: address,
+  });
+
+  const sortedProfiles = [...profiles].sort((a, b) => {
```
Comment (mdjastrzebski) [inline src/features/sign-in/SignInScreen.tsx:20]: [nit] consider using something like: https://lodash.com/docs/#sortBy

File: src/lib/lens-sdk/index.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import { LensConfig, development } from '@lens-protocol/react-native';
+import { bindings as wagmiBindings } from '@lens-protocol/wagmi';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+
+const lensConfig: LensConfig = {
+  bindings: wagmiBindings(),
+  // Use `development` or `production` depending on your need
+  environment: development,
+  // TODO: Check if security is an issue - IDEA: SecureStorage
+  storage: AsyncStorage,
```
Comment (mdjastrzebski) [inline src/lib/lens-sdk/index.ts:10]: As a follow up, pls do check if security is an issue here.

File: src/lib/navigation/NavIcon.tsx [hunk 7]
```diff
@@ -13,6 +13,8 @@ export function getIconForRoute(
   routeName: string,
 ): keyof typeof Octicons.glyphMap {
   switch (routeName) {
+    case 'SignIn':
+      return 'shield-lock';
     case 'Home':
       return 'apps';
     case 'Balance':
```
Comment (mdjastrzebski) [inline src/lib/navigation/NavIcon.tsx:20]: [nit] Check if we use all of these icons

**Review by mdjastrzebski — APPROVED** [review]
Looks good! I've marked some nit things.

One thing to take care after merging this PR is to verify the security implications of using AsyncStorage for Lens setup.

File: src/features/home/HomeScreen.tsx (hunk 3 above)
Comment (thiagobrez) [reply src/features/home/HomeScreen.tsx:10]: It's so small, won't it just add unnecessary complexity?

File: src/features/home/PublicationItem.tsx (hunk 4 above)
Comment (thiagobrez) [reply src/features/home/PublicationItem.tsx:41]: Good one, done!

File: src/features/sign-in/SignInScreen.tsx (hunk 5 above)
Comment (thiagobrez) [reply src/features/sign-in/SignInScreen.tsx:20]: IMO unnecessary to add a dependency to use one array method

File: src/lib/lens-sdk/index.ts (hunk 6 above)
Comment (thiagobrez) [reply src/lib/lens-sdk/index.ts:10]: Checked - changed to Expo SecureStorage

File: README.md (hunk 1 above)
Comment (thiagobrez) [reply README.md]: moved to Yarn to avoid having to use legacy deps

**Pull request merged**